category_theory/discrete_category.lean

```diff
@@ -0,0 +1,81 @@
+-- Copyright (c) 2017 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Stephen Morgan, Scott Morrison
+
+import data.ulift
+import category_theory.natural_transformation
+import category_theory.isomorphism
+import category_theory.functor_category
+
+namespace category_theory
+
+universes u₁ v₁ u₂ v₂
+
+def discrete (α : Type u₁) := α
+
+@[extensionality] lemma plift.ext {P : Prop} (a b : plift P) : a = b :=
```

Comment:
I can add a subsingleton instance, as
```
instance plift_subsingleton (P : Prop) : subsingleton (plift P) :=
subsingleton.intro (λ a b, begin cases a, cases b, refl end)
```
but I'm not actually sure what it would do for us. It doesn't seem like it's possible to define an `extensionality` lemma that uses a subsingleton instance. It would be great if `ext` could use subsingleton. 

Was it @cipher1024 who wrote `ext`? If so, do you have advice here?